languages/js/src/Polar.test.ts

```diff
@@ -802,3 +802,154 @@ describe('iterators', () => {
     ).toBe(6);
   });
 });
+
+// test_roles_integration
+describe('Polar roles', () => {
```

Comment:
I guess if this is easy enough to port seems reasonable (in reference to my comment in Notion)

Reply:
So is our plan to keep the tests in the host language libraries? Or are we moving testing into the core now that `enable_roles` is in the core?

Reply:
Either way--I assume we'll have a follow-up PR for either finishing porting the node tests or porting all tests to the core, right?

Reply:
There are four pieces of testing split into two main categories: validation & functionality.

@saolsen is currently working on splitting the validation tests in twain, pulling the bits we can test in the core into the core and leaving another chunk in the host language test suite. The latter will have to be ported to each new host language as we enable the roles feature.

For functionality tests, we want a single, exhaustive test suite (which would ideally live in the core) and then one test per host language library that makes sure the library-specific bits of the API work. Now that the policy lives in the core and is shared across languages, we decided there wasn't really much difference between pulling the Python test suite into the core and leaving it in Python — and the former would be significantly more work.

So, to answer your questions explicitly: we're keeping some tests in the host language libraries and putting some in the core, and the only follow-up PR will be #969 for validation tests, which Steve's working on.